core/src/main/scala/scalaz/Foldable.scala

```diff
@@ -325,6 +325,9 @@ trait Foldable[F[_]]  { self =>
         else seen
     }.reverse
 
+  def distinctBy[A, B: Equal](fa: F[A])(f: A => B): IList[A] =
+    distinctE(fa)(Equal.equalBy(f))
```

Comment:
Well no, it is the same, because the implementation of `Foldable.element` is 
```scala
def element[A: Equal](fa: F[A], a: A): Boolean = any(fa)(Equal[A].equal(a, _))
```
(also create a anonymous function)